default/tests/shared/test_annotation.py

```diff
@@ -35,6 +36,222 @@ def setUp(self):
+            inst.copy()
+        ]
+        video_data = {
+            'video_mode': True,
+            'video_file_id': file1.id,
+            'current_frame': frame.frame_number
+        }
+
+        ann_update = Annotation_Update(
+            session = self.session,
+            project = self.project,
+            video_data = video_data,
+            instance_list_new = instance_data,
+            file = file1,
+            do_init_existing_instances = True
+        )
+        result = ann_update.append_new_instance_list_hash(instance = instance1)
+        result2 = ann_update.append_new_instance_list_hash(instance = instance2)
+
+        self.assertTrue(result)
+        self.assertFalse(result2)
+
+    def test_order_new_instance_list_by_date(self):
+        file1 = data_mocking.create_file({'project_id': self.project.id, 'type': 'video'}, self.session)
+        frame = data_mocking.create_file(
+            {'project_id': self.project.id, 'type': 'frame', 'video_parent_file_id': file1.id, 'frame_number': 5}, self.session)
+        label_file = data_mocking.create_file({'project_id': self.project.id, 'type': 'label'}, self.session)
+        inst1 = {
+            'creation_ref_id': str(uuid.uuid4()),
+            'x_min': 1,
+            'y_min': 1,
+            'x_max': 18,
+            'y_max': 18,
+            'client_created_time': datetime.datetime(2020, 1, 1),
+            'soft_delete': False,
+            'label_file_id': label_file.id,
+            'type': 'box'
+        }
+        inst2 = {
+            'creation_ref_id': str(uuid.uuid4()),
+            'x_min': 1,
+            'y_min': 1,
+            'x_max': 18,
+            'y_max': 18,
+            'client_created_time': datetime.datetime(2020, 1, 2),
+            'soft_delete': False,
+            'label_file_id': label_file.id,
+            'type': 'box'
+        }
+        inst3 = {
+            'creation_ref_id': str(uuid.uuid4()),
+            'x_min': 1,
+            'y_min': 1,
+            'x_max': 18,
+            'y_max': 18,
+            'client_created_time': datetime.datetime(2020, 1, 3),
+            'soft_delete': False,
+            'label_file_id': label_file.id,
+            'type': 'box'
+        }
+        inst_list = [inst1, inst2, inst3]
+        ann_update = Annotation_Update(
+            session = self.session,
+            project = self.project,
+            video_data = None,
+            instance_list_new = inst_list,
+            file = file1,
+            do_init_existing_instances = True
+        )
+        ann_update.instance_list_new = inst_list
+        print('orders', ann_update.instance_list_new)
+        ann_update.order_new_instances_by_date()
+
+        self.assertEqual(ann_update.instance_list_new[0], inst3)
+        self.assertEqual(ann_update.instance_list_new[1], inst2)
+        self.assertEqual(ann_update.instance_list_new[2], inst1)
+
+
+    def test_special__removing_duplicate_instances_in_new_instance_list(self):
```

Comment:
@PJEstrada  GREAT!!! :)   🎉
And that's a great used of added counts! glad to see that working in there

Reply:
Thanks Anthony!! :)